src/argprocessing.cpp

```diff
@@ -262,743 +263,755 @@ process_arg(const Context& ctx,
 {
   size_t& i = args_index;
 
-  // The user knows best: just swallow the next arg.
-  if (args[i] == "--ccache-skip") {
-    i++;
-    if (i == args.size()) {
-      LOG_RAW("--ccache-skip lacks an argument");
-      return Statistic::bad_compiler_arguments;
+  if (!state.found_end_of_options) {
```

Comment:
I would prefer not adding this extra indention level. I'll refactor the code to make it easier to do the change you propose.

Reply:
Done in 62f3217ee9f2b3b147ad9b31fc317b10e9dfc225.